libraries/state_history/include/eosio/state_history/log.hpp

```diff
@@ -119,9 +132,37 @@ class state_history_log {
             vacuum();
          }
       }
+
+       thr = std::thread([this] {
```

Comment:
An `eosio::chain::named_thread_pool` (even with a size of 1) may allow you to remove some of this boilerplate. Not critical, judgement up to you, just pointing it out.

Reply:
eh the whole `write_thread_has_exception` thing may torpedo that, dunno. Still worth thinking over.